modules/porous_flow/tests/fluidstate/theis_brineco2.i

```diff
@@ -0,0 +1,289 @@
+  [./pgas]
+    initial_condition = 20e6
+  [../]
+  [./zi]
+    initial_condition = 0
+  [../]
+[]
+
+[Kernels]
+  [./mass0]
+    type = PorousFlowMassTimeDerivative
+    fluid_component = 0
+    variable = pgas
+  [../]
+  [./flux0]
+    type = PorousFlowAdvectiveFlux
+    fluid_component = 0
+    variable = pgas
+  [../]
+  [./mass1]
+    type = PorousFlowMassTimeDerivative
+    fluid_component = 1
+    variable = zi
+  [../]
+  [./flux1]
+    type = PorousFlowAdvectiveFlux
+    fluid_component = 1
+    variable = zi
+  [../]
+[]
+
+[UserObjects]
+  [./dictator]
+    type = PorousFlowDictator
+    porous_flow_vars = 'pgas zi'
+    number_fluid_phases = 2
+    number_fluid_components = 2
+  [../]
+[]
+
+[Modules]
+  [./FluidProperties]
+    [./co2]
+      type = CO2FluidProperties
+    [../]
+    [./brine]
+      type = BrineFluidProperties
+    [../]
+  [../]
+[]
+
+[Materials]
+  [./temperature]
+    type = PorousFlowTemperature
+    at_nodes = true
+  [../]
+  [./temperature_qp]
+    type = PorousFlowTemperature
+  [../]
+  [./brineco2]
+    type = PorousFlowFluidStateBrineCO2
+    gas_porepressure = pgas
+    z = zi
+    co2_fp = co2
+    brine_fp = brine
+    at_nodes = true
+    temperature_unit = Celsius
+    xnacl = xnacl
+    sat_lr = 0.1
+  [../]
+  [./brineco2_qp]
+    type = PorousFlowFluidStateBrineCO2
+    gas_porepressure = pgas
+    z = zi
+    co2_fp = co2
+    brine_fp = brine
+    temperature_unit = Celsius
+    xnacl = xnacl
+    sat_lr = 0.1
+  [../]
+  [./porosity]
+    type = PorousFlowPorosityConst
+    at_nodes = true
+    porosity = 0.2
+  [../]
+  [./permeability]
+    type = PorousFlowPermeabilityConst
+    permeability = '1e-12 0 0 0 1e-12 0 0 0 1e-12'
+  [../]
+  [./relperm_water]
+    type = PorousFlowRelativePermeabilityCorey
+    at_nodes = true
+    n = 2
+    phase = 0
+    s_res = 0.1
+    sum_s_res = 0.1
+  [../]
+  [./relperm_gas]
+    type = PorousFlowRelativePermeabilityCorey
+    at_nodes = true
+    n = 2
+    phase = 1
+  [../]
+  [./relperm_all]
+    type = PorousFlowJoiner
+    at_nodes = true
+    material_property = PorousFlow_relative_permeability_nodal
+  [../]
+[]
+
+[BCs]
+  [./rightwater]
+    type = DirichletBC
+    boundary = right
+    value = 20e6
+    variable = pgas
+  [../]
+[]
+
+[DiracKernels]
+  [./source]
+    type = PorousFlowSquarePulsePointSource
+    point = '0 0 0'
+    mass_flux = 2
+    variable = zi
+  [../]
+[]
+
+[Preconditioning]
+  [./smp]
+    type = SMP
+    full = true
+    petsc_options = '-snes_converged_reason -ksp_diagonal_scale -ksp_diagonal_scale_fix -ksp_gmres_modifiedgramschmidt -snes_linesearch_monitor'
+    petsc_options_iname = '-ksp_type -pc_type -sub_pc_type -sub_pc_factor_shift_type -pc_asm_overlap -snes_atol -snes_rtol -snes_max_it'
+    petsc_options_value = 'gmres      asm      lu           NONZERO                   2               1E-8       1E-10 20'
+  [../]
+[]
+
+[Executioner]
+  type = Transient
+  solve_type = NEWTON
+  end_time = 1e5
```

Comment:
That's fine - as @andrsd pointed out - I failed to notice that the test was market heavy.  It's fine for heavy tests to be longer.